app/models/smart_proxy.rb

```diff
@@ -100,6 +100,14 @@ def setting(feature, setting)
     smart_proxy_features.find_by(:feature_id => Feature.find_by(:name => feature)).try(:settings).try(:[], setting)
   end
 
+  def httpboot_http_port
+    setting(:HTTPBoot, 'http_port') || raise(::Foreman::Exception.new(N_("HTTP boot requires proxy with httpboot feature and http_port exposed setting")))
```

Comment:
Do we need to check if the proxy has the appropriate feature?

Reply:
IIRC the data model is that the settings are stored in the Feature model. So having a setting implies the feature. It may be useful to check it separate when you want to raise more explicit exceptions. I can imagine a `FeatureMissingException` or something. However, that's not a requirement for this PR.

Reply:
The code actually does this. Method `settings` uses `try`.